stan/math/rev/mat/functor/cvodes_ode_data.hpp

```diff
@@ -150,11 +150,11 @@ class cvodes_ode_data {
    */
   inline void rhs(double t, const double y[], double dy_dt[]) const {
     const std::vector<double> y_vec(y, y + N_);
-    const std::vector<double> dy_dt_vec
+    const std::vector<double>& dy_dt_vec
         = f_(t, y_vec, theta_dbl_, x_, x_int_, msgs_);
     check_size_match("cvodes_ode_data", "dz_dt", dy_dt_vec.size(), "states",
                      N_);
-    std::copy(dy_dt_vec.begin(), dy_dt_vec.end(), dy_dt);
+    std::move(dy_dt_vec.begin(), dy_dt_vec.end(), dy_dt);
```

Comment:
Could you explain why you are changing `std::copy` to `std::move`? This is not quite clear to me (based on comments and description of the PR). Is there a difference between the two if `dy_dt` is a vector of primitives?

Reply:
This comment applies to all the places where `std::copy` has been changed to `std::move`.

Reply:
The intention is to allow the compiler further optimisations here. If we copy then we tell the compiler that we want two distinct copies... which we do not need. Changing this to a move tells the compiler that we don't need the stuff in the original structure any more. This way the compiler can hopefully inline better and avoid a few copies.

Reply:
I get that. I'm merely suggestion that the compiler already does all the optimization -- the resulting machine language code should be the same. Of course, it's not easy to test; recently I've started paying with https://godbolt.org/ to look into these things.

In any case, the new code is at least better than the old one, and for what it's worth, give us a better sense of what is going on. So I'm happy with the current commit.